### PR TITLE
Feature/lp 114 repair request viewmodel

### PR DESCRIPTION
**Pull request by JoseGR98**
LP-114   Repair request - ViewModel
https://devaptivist.atlassian.net/browse/LP-114?atlOrigin=eyJpIjoiMTlmZDc0MzlhMGUyNDE2NjhlNTA3YmUyN2ZjMjc3Y2QiLCJwIjoiaiJ9

ViewModel worked for the RepairView. Several changes:
Classes:
RepairModel --> rename to RepairRequest
UIEventsRepair --> to handle events type

Navigation:
ApartmentGraph --> small changes

RepairUI:
RepairViewModel --> big changes, added events to handle data manipulation
RepairView --> small changes, adapted to Events

### CODE REVIEW

File: core/data/src/main/java/com/iteneum/RepairModel.kt
Comment (melissaquezada94) [inline core/data/src/main/java/com/iteneum/RepairModel.kt]: Usually, the data class's attributes are of val type, we declare them var if it is really necessary.

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: The repair model could ve null?

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: You are not using this variable: `dataInfo` to show the information when the view charged

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: You are not observing the contatPhone when it changes

File: core/data/src/main/java/com/iteneum/RepairModel.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/RepairModel.kt]: Okay, got it! Changed! :)

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: For the moment it will be like that.
Based in Sergio & Carlos comments, he instructed to add the Unit to the RepairView, since it has to show the Department/House number. Because of that, it will be mostly probable that the View will have two type of perspectives. For the Employer, to be able to manipulate the field; and for the employee, to see the value already inserted. In case that it cannot be attached or found, we have to enable the null possibility.
What I will add for now is a TODO to verify this later on.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Yes, you are right. But I wrote on the TODO:
/* TODO - LpOutlinedTextFieldInput (Check component required) - To enable to action of passing value argument --> value = dataInfo?.unitDepartment ?: "000" */

There is a problem related to obtain the Unit (“Department number”) from the data source and paste it on the Repair View (as desired by Sergio). The component used to have the Unit does not have the actual “value” parameter. We need to add it so we can use it. This involves however the modification of the actual component.

The parameter value is not on the function parameters. It is required to be able to pass and use the dataInfo.
image

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: You are updating the wrong attibute plese check all

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: I did not get this point, sorry. I thougth that is showing correctly. Because i have it on two parameters.

value (So it can print the phoneNumber when inserting/deleting a number
image

onPhoneChanged (So it can be updated when the user inserts another value)
image

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: That was a typo, my apologies. Corrected!

image

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (melissaquezada94) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: You can put your task in block status because we are working on these changes to set the value to the input. 
Another thing is why are you using this parameter: ` dataInfo: RepairModel?,` because the only use that I see is to show the unit deparmet and if you are using only for this so I recommend you to change the name variable to `unitDepartment: String`

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (melissaquezada94) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Ok so this case is only for the attribute unitDartment, right? So in this case why don't you send just the variable unitDepartment: String? instead all the data class

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Why are you using a default RepairModel? My question is because when I navigate to your view I'm not able to see all this values in the corresponding inputs

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (melissaquezada94) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:64]: In this function, I don't see any remember for each input value. We need a remember for each variable being observed from the viewModel to display the information that we get from the server( in this case is mocked) and also to be updated. This is because the information should be getting and updating from the viewModel and the View just need to show or call a updated method from the view and the information should be synchronized. 
Like in this case mostly of the components have it owns remember, I talk to Irving to change this so you can start working on add a remember of each variable that it is observer by the viewModel.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (melissaquezada94) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Well, I mean, that you need to create an remember in this way: 
viewModel: 
 `private val _uiState = mutableStateOf(UiStateMainActivity(data = test, loading = true))
    val uiState: State<UiStateMainActivity> = _uiState`
    
    `fun updateLoading(){
        _uiState.value = _uiState.value.copy(
            loading = !_uiState.value.loading
        )
    }`
Compose: 
`private val viewModel by viewModels<MainActivityViewModel>()`
`val uiState by remember { viewModel.uiState }`
`Button(
                        onClick = { viewModel.updateLoading() },
                        modifier = Modifier.padding(bottom = 20.dp)
                    ) {
                        Text(text = "Click ${uiState.loading}")
                    }`

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: I agree. Ok, will change to blocked and also make the change of unitDepartment.

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Agree. I will make the change

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, will make the change. Thank you!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: I made the change that instead of creating it null, I will make an initialization of the RepairModel data class. Is this better?

New declaration
image

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: You are right, I only use the "unitDeparment" value, so will reduce it to that only.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:64]: Could you give me a structure example? Because since I am using compose I thought that the observers were this methods on file "RepairViewModel"

image


Or do you want something like this other example like when using Live Data for observables?
image

I don't know a lot of this yet, so your comments are really helpful. Thank you!

File: core/data/src/main/java/com/iteneum/RepairModel.kt
Comment (irv205) [inline core/data/src/main/java/com/iteneum/RepairModel.kt]: Rename to RepairRequest

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (irv205) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Rename to "repairRequest"

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: if we are using a mutbleStateOf we cand change the implementation like

var  myData: RepairModel by mutableStateOf(RepairModel)
        private set

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (irv205) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: What is the status?

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (irv205) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:64]: What is the status?

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (irv205) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Status?

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: We don't need send the viewModel to a second composable view, viewModel always must be in the first one composable view, if you want to send a value or an object to the second one you need to access to that value and then add a new param to the second composable function.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we need to use hight order function here and send the event to the top

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we need to use hight order function here and send the event to the top

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we need to use hight order function here and send the event to the top

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we need to use hight order function here and send the event to the top

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (carlosph816) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: we need to use hight order function here and send the event to the top

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Done! Uploaded already

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Not sure of the meaning of this one. I would like to ask for assistance @irv205

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Change made but pending for new comments

File: core/data/src/main/java/com/iteneum/RepairModel.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/RepairModel.kt]: Ok, done!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Ok, done as well!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Ok done! Thanks for the explanation in person as well

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:64]: Not sure about this one. I thought that _remember_ is not required anymore since the "ViewModel" is the one that handles that data. As I understood, the View does not require to remember any value. @irv205 
Therefore, I do not see the necessity of adding remember here.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, modified! Aided by you, thank you very much!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Modified! Thank you for your help

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Modified! Thank you for your help

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Modified! Thank you for your help

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Modified! Thank you for your help

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Modified. Done! (Help provided by Carlos)

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt:64]: ViewModel reworked with events. Aided by Carlos. Done!

**Comment by JoseGR98** [comment]
Waiting for Code Review, during next week. @irv205 
Waiting for Code Review at May 15th 2023

**Review by carlosph816 — APPROVED** [review]

**Comment by JoseGR98** [comment]
1 code reviewer approved. Waiting for one more or new feedback. I will need your approval @irv205

File: core/data/src/main/java/com/iteneum/UIEventRepair.kt
Comment (Lupillo17) [inline core/data/src/main/java/com/iteneum/UIEventRepair.kt]: Should be only 1 author, change it for modifiedBy

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:22]: Should be only 1 author, change it for modifiedBy

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:34]: line 31, add remining states for response, acctually is just success state, you should add error state and loading state

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Move RepairContainer documentation to their respective function.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: move arrayOf to the viewModel and retrive it as value.

File: core/data/src/main/java/com/iteneum/RepairRequest.kt
Comment (Lupillo17) [inline core/data/src/main/java/com/iteneum/RepairRequest.kt:1]: By the moment move this file to **feature/repair** package, not in **core package**

File: core/data/src/main/java/com/iteneum/UIEventRepair.kt
Comment (Lupillo17) [inline core/data/src/main/java/com/iteneum/UIEventRepair.kt:1]: By the moment move this file to **feature/repair** package, not in **core package**

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:86]: add missing parameters to **RepairRequest** object

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: I think **LeasePertTheme** is not necesary, try delete it and it should works well

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: I can't scroll the view, please add vertical scrolling to **Column Modifier**

File: core/data/src/main/java/com/iteneum/UIEventRepair.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/UIEventRepair.kt]: Ok, removed!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:22]: Ok, updated :)

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:34]: Ok, added!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, moved bellow

File: core/data/src/main/java/com/iteneum/RepairRequest.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/RepairRequest.kt:1]: Ok, moved!

File: core/data/src/main/java/com/iteneum/UIEventRepair.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/UIEventRepair.kt:1]: Ok, moved!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt:86]: Ok. They are not required because user will fill them, but added.

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, removed it!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, added!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, done!

**Comment by JoseGR98** [comment]
Changed updated! Waiting for new code approbal @Lupillo17

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: By the moment add a TODO and delete line 39

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: By the moment add a TODO and delete line 42

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: check orthography **thw phone**

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Describe the main pourpose of RepairContainer, for example: contains the view of repair screen... describe it

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Ok, done!

File: feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/data/RepairViewModel.kt]: Ok,done!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Ok, corrected!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Done!

**Comment by JoseGR98** [comment]
New PR uploaded , addressed @Lupillo17

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: check orthography

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (Lupillo17) [inline feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: check description and correct it please

**Review by Lupillo17 — APPROVED** [review]

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Corrected!

File: feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt
Comment (JoseGR98) [reply feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt]: Checked and improved!

**Pull request merged**
